drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -5002,6 +5264,217 @@ RDD::PipelineID RenderingDeviceDriverVulkan::render_pipeline_create(
+	// Vertex positions is first buffer
+	const VertexFormatInfo *vf_info = (const VertexFormatInfo *)p_vertex_format.id;
+	VkDeviceSize buffer_offset = vf_info->vk_attributes[0].offset;
+
+	VkDeviceAddress vertex_address = _buffer_get_device_address(p_vertex_buffer) + buffer_offset;
+	uint32_t index_size = sizeof(uint16_t);
+	VkDeviceAddress index_address = _buffer_get_device_address(p_index_buffer) + p_index_offset * index_size;
+	VkDeviceAddress transform_address = _buffer_get_device_address(p_transform_buffer) + p_transform_offset;
+
+	VkDeviceSize vertex_stride = vf_info->vk_bindings[0].stride;
+	VkFormat vertex_format = vf_info->vk_attributes[0].format;
+	uint32_t max_vertex = p_vertex_count ? p_vertex_count - 1 : 0;
+
+	AccelerationStructureInfo *accel_info = VersatileResource::allocate<AccelerationStructureInfo>(resources_allocator);
+
+	accel_info->geometry.sType = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_KHR;
+	accel_info->geometry.geometryType = VK_GEOMETRY_TYPE_TRIANGLES_KHR;
+	accel_info->geometry.flags = VK_GEOMETRY_OPAQUE_BIT_KHR;
+
+	accel_info->geometry.geometry.triangles.sType = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_TRIANGLES_DATA_KHR;
+	accel_info->geometry.geometry.triangles.vertexFormat = vertex_format;
+	accel_info->geometry.geometry.triangles.vertexData.deviceAddress = vertex_address;
+	accel_info->geometry.geometry.triangles.vertexStride = vertex_stride;
+	accel_info->geometry.geometry.triangles.indexType = p_index_format == INDEX_BUFFER_FORMAT_UINT16 ? VK_INDEX_TYPE_UINT16 : VK_INDEX_TYPE_UINT32;
+	accel_info->geometry.geometry.triangles.indexData.deviceAddress = index_address;
+	// Transform matrix, 3 rows, 4 columns, row-major
+	accel_info->geometry.geometry.triangles.transformData.deviceAddress = transform_address;
+	// Number of vertices in vertexData minus one, aka max vertex index
+	accel_info->geometry.geometry.triangles.maxVertex = max_vertex;
+
+	// Info for building BLAS
+	uint32_t primitive_count = (p_vertex_count - p_vertex_offset) / 3;
+	if (p_index_buffer) {
+		primitive_count = p_index_count / 3;
+	}
+	accel_info->range_info.firstVertex = p_vertex_offset;
+	accel_info->range_info.primitiveCount = primitive_count;
+	accel_info->range_info.primitiveOffset = 0;
+	accel_info->range_info.transformOffset = 0;
+	uint32_t max_primitive_count = accel_info->range_info.primitiveCount;
+
+	accel_info->build_info.sType = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_BUILD_GEOMETRY_INFO_KHR;
+	accel_info->build_info.type = VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR;
+	accel_info->build_info.mode = VK_BUILD_ACCELERATION_STRUCTURE_MODE_BUILD_KHR;
+	accel_info->build_info.pGeometries = &accel_info->geometry;
+	accel_info->build_info.geometryCount = 1;
+	accel_info->build_info.flags = VK_BUILD_MICROMAP_PREFER_FAST_TRACE_BIT_EXT;
```

Comment:
This should be `VK_BUILD_ACCELERATION_STRUCTURE_PREFER_FAST_TRACE_BIT_KHR` instead; I'd guess validation layers should also complain about the use of micromap bits without the enablement of the relevant extension.